seed/analysis_pipelines/pipeline.py

```diff
@@ -31,49 +31,166 @@ def task_create_analysis_property_views(analysis_id, property_view_ids, progress
+            try:
+                analysis = Analysis.objects.get(id=_analysis_id)
+                if analysis.status == expected_status:
+                    pass
+                elif analysis.status in [Analysis.STOPPED, Analysis.FAILED]:
+                    # assume someone else stopped or failed the analysis and that we shouldn't run the task
+                    log_message = {
+                        'analysis_id': _analysis_id,
+                        'debug_message': 'Analysis has already been stopped or failed before starting the task. '
+                                         'Not running the task and stopping the task chain.'
+                    }
+                    logger.info(json.dumps(log_message))
+                    _stop_task_chain(_self)
+                    return
+                else:
+                    # something Bad has happened
+                    raise AnalysisPipelineException(f'When preparing to run the task {func}, expected analysis status to be {expected_status} but it was {analysis.status}')
+            except Analysis.DoesNotExist:
+                # someone deleted the analysis
+                log_message = {
+                    'analysis_id': _analysis_id,
+                    'debug_message': 'Analysis no longer exists before starting the task. '
+                                     'Assuming the analysis was deleted and stopping the celery task chain.'
+                }
+                logger.info(json.dumps(log_message))
+                _stop_task_chain(_self)
+                return
+
+            # Catch all exceptions raised by the task
+            # If the exception was to stop the task chain, stop the task chain
+            # If the analysis no longer exists, ignore the exception (the analysis was deleted by someone else)
+            # If the analysis _does_ still exist, raise the exception (something unexpected happened)
+            try:
+                return func(*args, **kwargs)
+            except StopAnalysisTaskChain as e:
+                log_message = {
+                    'analysis_id': _analysis_id,
+                    'debug_message': 'StopAnalysisTaskChain exception raised, stopping the celery task chain.',
+                    'exception': repr(e)
+                }
+                logger.info(json.dumps(log_message))
+                _stop_task_chain(_self)
+                return
+            except Exception as e:
+                try:
+                    Analysis.objects.get(id=_analysis_id)
+
+                    log_message = {
+                        'analysis_id': _analysis_id,
+                        'debug_message': 'Caught unexpected exception occurred during analysis task (and analysis still exists). Re-raising exception and continuing.',
+                        'exception': repr(e)
+                    }
+                    logger.error(json.dumps(log_message))
+                    raise e
```

Comment:
I'm likely misunderstanding something, but I would've expected us to create an AnalysisMessage in here somewhere. Does that happen else where?

Reply:
I think that's a good idea. I will add that with a generic user message saying something unexpected went wrong.